.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,65 @@
+name: Deploy functions
+
+on:
+  workflow_call: {}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 3
+
+      - name: Build functions
+        uses: "./.github/actions/build"
+
+      - name: Make the azure function app artifact
+        id: make_artifact
+        run: |
+          zip -r io-functions-service-messages.zip .
+          echo "artifact-path=$(realpath io-functions-service-messages.zip)" >> "$GITHUB_OUTPUT"
+        shell: bash
+      
+      - name: Upload functions artifact
+        uses: actions/upload-artifact@v3
+        with:
+          path: ${{ steps.make_artifact.outputs.artifact-path }}
+
+  deploy:
+    name: Deploy
+    if: ${{ !github.event.act }}
+    needs: [build]
+    runs-on: self-hosted
+    environment: production
```

Comment:
`prod-cd` but since this is a `workflow_call` it'd better to accept this value from inputs like this imho:
```yaml
inputs:
  environment:
    type: string

deploy:
  environment: ${{ inputs.environment }}-cd
```